databricks/koalas/tests/test_groupby.py

```diff
@@ -182,6 +182,41 @@ def test_diff(self):
         self.assert_eq(repr(kdf.groupby(['b'])['a'].diff().sort_index()),
                        repr(pdf.groupby(['b'])['a'].diff().sort_index()))
 
+    def test_fillna(self):
+        pdf = pd.DataFrame({'A': [1, 1, 2, 2],
+                            'B': [2, 4, None, 3],
+                            'C': [None, None, None, 1],
+                            'D': [0, 1, 5, 4]}, columns=['A', 'B', 'C', 'D'])
+        kdf = koalas.DataFrame(pdf)
+        self.assert_eq(kdf.groupby("A").fillna(0),
+                       pdf.groupby("A").fillna(0))
+        self.assert_eq(kdf.groupby("A").fillna(method='bfill'),
+                       pdf.groupby("A").fillna(method='bfill'))
+        self.assert_eq(kdf.groupby("A").fillna(method='ffill'),
+                       pdf.groupby("A").fillna(method='ffill'))
+
+    def test_ffill(self):
+        pdf = pd.DataFrame({'A': [1, 1, 2, 2],
+                            'B': [2, 4, None, 3],
+                            'C': [None, None, None, 1],
+                            'D': [0, 1, 5, 4]}, columns=['A', 'B', 'C', 'D'])
+        kdf = koalas.DataFrame(pdf)
+        self.assert_eq(kdf.groupby("A").ffill(),
+                       pdf.groupby("A").ffill())
+        self.assert_eq(repr(kdf.groupby("A")['B'].ffill()),
+                       repr(pdf.groupby("A")['B'].ffill()))
+
+    def test_bfill(self):
+        pdf = pd.DataFrame({'A': [1, 1, 2, 2],
+                            'B': [2, 4, None, 3],
+                            'C': [None, None, None, 1],
+                            'D': [0, 1, 5, 4]}, columns=['A', 'B', 'C', 'D'])
+        kdf = koalas.DataFrame(pdf)
+        self.assert_eq(kdf.groupby("A").bfill(),
+                       pdf.groupby("A").bfill())
+        self.assert_eq(repr(kdf.groupby("A")['B'].bfill()),
+                       repr(pdf.groupby("A")['B'].bfill()))
```

Comment:
```
E   AssertionError: False is not true : DataFrames are not equal: 
E   
E   Left:
E        B    C  D
E   0  2.0  NaN  0
E   1  4.0  NaN  1
E   2  3.0  1.0  5
E   3  3.0  1.0  4
E   B    float64
E   C    float64
E   D      int64
E   dtype: object
E   
E   Right:
E      A    B    C  D
E   0  1  2.0  NaN  0
E   1  1  4.0  NaN  1
E   2  2  3.0  1.0  5
E   3  2  3.0  1.0  4
E   A      int64
E   B    float64
E   C    float64
E   D      int64
E   dtype: object
```

Looks pandas DataFrame in lower versions has additional column. You can drop this column from `pdf` and add some comments so that the tests can pass in lower versions too.

For instance, 

```python
if 'a' in pdf.columns:
    # In low pandas versions, it returns columns 'a' whereas higher version does not. Koalas follows
    # latest pandas versions so it drops column 'a' to test with low pandas versions as well.
    pdf = pdf.drop(columns=["a"])
```

Reply:
Or,

```python
if LooseVersion("0.25") > LooseVersion(pd.__version__):
    # In pandas versions lower than 0.25, it returns columns 'a' whereas 0.25 version does not.
    # Koalas follows latest pandas versions so it drops column 'a' to test with low pandas
    # versions as well.
    pdf = pdf.drop(columns=["a"])

...

self.assert_eq(...)
```

Reply:
I'd prefer to use `LooseVersion` in this case.